### PR TITLE
Added a .clang-format configuration file to enforce the project's con…

### PR DESCRIPTION
**Pull request by dkozinski**
…sistent coding style

Example usage: clang-format -i *.cpp *.h

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+AllowShortLoopsOnASingleLine: false
+
+# Space settings
+SpaceBeforeParens: Custom
+SpaceBeforeParensOptions:
+  AfterControlStatements: false
+  AfterFunctionDefinitionName: false
+
+SpaceAfterCStyleCast: true
+#SpaceBeforeCStyleCast: true
```
Comment (dariusz-f) [inline .clang-format:37]: Do not comment line, just switch to false if it required.

**Comment by kpchoi** [comment]
@dariusz-f 
I have simple question.
After applying this clang-format, what will the existed code happen?
Is this formatting only for PR files (committed file)?
or all files will be aligned to this format?

**Comment by dariusz-f** [comment]
This file is only rule set for formatting. It will not change codebase. 

In next step we will prepare GitHub action that will use this rules to check all future PR's and merges for styling.

Existing code have to be refactored manually to check and apply new styling rules. This also will be done in a separete PR.

**Review by dariusz-f — APPROVED** [review]

**Comment by kpchoi** [comment]
@dariusz-f 
I have simple question.
After applying this clang-format, what will the existed code happen?
Is this formatting only for PR files (committed file)?
or all files will be aligned to this format?

File: .clang-format (hunk 1 above)
Comment (kpchoi) [reply .clang-format]: I found that the SpaceBeforeCStyleCast is not supported in clang-format
```
Error reading ../.clang-format: Invalid argument
../.clang-format:37:1: error: unknown key 'SpaceBeforeCStyleCast'
SpaceBeforeCStyleCast: false
^~~~~~~~~~~~~~~~~~~~~
```

**Comment by kpchoi** [comment]
Hi @dariusz-f and @dkozinski ,
When I tested current clang-format,
it is changing array initializing code a lot.
Could you make it 'no change?'
I am thinking that clang-format is introducing too many changes.
How can we make a rule with minimized changing?
1) use 4 space for indent
2) if/else brace: no wrapping.

That's all.

**Comment by kpchoi** [comment]
@dkozinski @dariusz-f 
After testing some code with clang-format, 
I found that it changes too much aggressively.
I think that sometimes manual formating is the best way only if we consider human-mistake.
For example, 4-step space indent can be mistaken to 1 tab, according to IDE tool settings.
Is there any what only such simple mistake can be announced by GItHub automation?

**Comment by dkozinski** [comment]
Hi @kpchoi ,

To prevent clang-format from changing array initialization, we can use the AlignConsecutiveDeclarations option in our .clang-format configuration file.
However, clang-format may still apply other formatting rules, so we might need to experiment with additional options depending on our formatting needs.

Another way to preserve specific formatting for certain lines or blocks of code, is using clang-format's // clang-format off and // clang-format on comments to disable formatting for those sections:

// clang-format off
int array[] = {1, 2, 3, 4, 5, 6, 7, 8};
// clang-format on

Would this way be acceptable? 
We will experiment with various clang-format options to achieve the desired effect.

**Comment by dkozinski** [comment]
We can disable specific formatting. If we want to prevent Clang format from changing specific parts of our code, we can use comments to disable formatting for that section, as follows:


// clang-format off
```
const u16 tbl_scan[BLOCK_D] =
{
    0,    1,    8,   16,    9,    2,    3,   10,
    17,   24,   32,   25,   18,   11,    4,    5,
    12,   19,   26,   33,   40,   48,   41,   34,
    27,   20,   13,    6,    7,   14,   21,   28,
    35,   42,   49,   56,   57,   50,   43,   36,
    29,   22,   15,   23,   30,   37,   44,   51,
    58,   59,   52,   45,   38,   31,   39,   46,
    53,   60,   61,   54,   47,   55,   62,   63,
};
// clang-format on
```

**Comment by mpeg5** [comment]
Hi @dkozinski @dariusz-f ,
I believe that we could merge this .clang-format file into xeve project to understand coding style of xeve.
But, I'd like to suggest to postpone to deploy clang-format to Github action because it can lead too many changes in the code and the code is not ready to prevent that situation.

**Comment by dkozinski** [comment]
> This is another example; Original;
> 
> ```
> const u16 tbl_scan[BLOCK_D] =
> {
>     0,    1,    8,   16,    9,    2,    3,   10,
>     17,   24,   32,   25,   18,   11,    4,    5,
>     12,   19,   26,   33,   40,   48,   41,   34,
>     27,   20,   13,    6,    7,   14,   21,   28,
>     35,   42,   49,   56,   57,   50,   43,   36,
>     29,   22,   15,   23,   30,   37,   44,   51,
>     58,   59,   52,   45,   38,   31,   39,   46,
>     53,   60,   61,   54,   47,   55,   62,   63,
> };
> ```
> 
> clang-format changed to ;
> 
> ```
> const u16 tbl_scan[BLOCK_D] =
>     {
>         0,
>         1,
>         8,
>         16,
>         9,
>         2,
>         3,
>         10,
>         17,
>         24,
>         32,
>         25,
>         18,
>         11,
>         4,
>         5,
>         12,
>         19,
>         26,
>         33,
>         40,
>         48,
>         41,
>         34,
>         27,
>         20,
>         13,
>         6,
>         7,
>         14,
>         21,
>         28,
>         35,
>         42,
>         49,
>         56,
>         57,
>         50,
>         43,
>         36,
>         29,
>         22,
>         15,
>         23,
>         30,
>         37,
>         44,
>         51,
>         58,
>         59,
>         52,
>         45,
>         38,
>         31,
>         39,
>         46,
>         53,
>         60,
>         61,
>         54,
>         47,
>         55,
>         62,
>         63,
> };
> ```



The issue is caused by the trailing comma immediately before the closing brace.
Adding a trailing comma (which is optional in C) after the last array element causes clang-format to put each array's element on a separate line.

**Comment by cpncf** [comment]
Hi @dkozinski @dariusz-f ,
I believe that we could merge this .clang-format file into xeve project to understand coding style of xeve.
But, I'd like to suggest to postpone to deploy clang-format to Github action because it can lead too many changes in the code and the code seems to be not ready to prevent that situation.

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+# Space settings
+SpaceBeforeParens: Custom
+SpaceBeforeParensOptions:
+  AfterControlStatements: false
+
+# Other settings
+ColumnLimit: 160  # Disable column limit
+
+# Put each one of the finction paremeters on its own line.
```
Comment (dariusz-f) [inline .clang-format:42]: "finction" typo

File: .gitignore
Comment (dariusz-f) [inline .gitignore:1]: Do not include this file in this PR

**Review by dariusz-f — CHANGES_REQUESTED** [review]

**Review by dariusz-f — APPROVED** [review]

**Pull request merged**